### PR TITLE
`unittest.case`: Further stub improvements

### PR DESCRIPTION
**Pull request by AlexWaygood**
Fixes several things I noticed while preparing #8066.

- Use `object` instead of `Any` where literally any argument can be safely passed in.
- Where the runtime uses assignments for method aliases, do the same thing in the stub.
- Give a more precise type for `assertGreater`, `assertLess`, `assertGreaterEqual` and `assertLessEqual`.
- Correct the outdated type for `assert(Not)IsInstance`.

### CODE REVIEW

File: stdlib/unittest/case.pyi [hunk 1]
```diff
@@ -105,18 +108,18 @@ class TestCase:
...
     def assertIn(self, member: Any, container: Iterable[Any] | Container[Any], msg: Any = ...) -> None: ...
     def assertNotIn(self, member: Any, container: Iterable[Any] | Container[Any], msg: Any = ...) -> None: ...
-    def assertIsInstance(self, obj: Any, cls: type | tuple[type, ...], msg: Any = ...) -> None: ...
-    def assertNotIsInstance(self, obj: Any, cls: type | tuple[type, ...], msg: Any = ...) -> None: ...
-    def assertGreater(self, a: Any, b: Any, msg: Any = ...) -> None: ...
-    def assertGreaterEqual(self, a: Any, b: Any, msg: Any = ...) -> None: ...
-    def assertLess(self, a: Any, b: Any, msg: Any = ...) -> None: ...
-    def assertLessEqual(self, a: Any, b: Any, msg: Any = ...) -> None: ...
+    def assertIsInstance(self, obj: object, cls: _IsInstanceClassInfo, msg: Any = ...) -> None: ...
+    def assertNotIsInstance(self, object: Any, cls: _IsInstanceClassInfo, msg: Any = ...) -> None: ...
```
Comment (AlexWaygood) [inline stdlib/unittest/case.pyi:118]:
```suggestion
    def assertNotIsInstance(self, obj: object, cls: _IsInstanceClassInfo, msg: Any = ...) -> None: ...
```

File: stdlib/builtins.pyi [hunk 2]
```diff
@@ -1321,17 +1322,12 @@ def iter(__function: Callable[[], _T], __sentinel: object) -> Iterator[_T]: ...
...
-    ) -> bool: ...
-    def issubclass(
-        __cls: type, __class_or_tuple: type | types.UnionType | tuple[type | types.UnionType | tuple[Any, ...], ...]
-    ) -> bool: ...
-
+    _IsInstanceClassInfo: TypeAlias = Union[type, types.UnionType, tuple[type | types.UnionType | tuple[Any, ...], ...]]
 else:
-    def isinstance(__obj: object, __class_or_tuple: type | tuple[type | tuple[Any, ...], ...]) -> bool: ...
-    def issubclass(__cls: type, __class_or_tuple: type | tuple[type | tuple[Any, ...], ...]) -> bool: ...
+    _IsInstanceClassInfo: TypeAlias = type | tuple[type | tuple[Any, ...], ...]
```
Comment (AlexWaygood) [inline stdlib/builtins.pyi:1327]:
```suggestion
    _IsInstanceClassInfo: TypeAlias = Union[type, tuple[type | tuple[Any, ...], ...]]
```

**Comment by AlexWaygood** [comment]
> Diff from [mypy_primer](https://github.com/hauntsaninja/mypy_primer), showing the effect of this PR on open source code:
> 
> ```diff
> tornado (https://github.com/tornadoweb/tornado)
> + tornado/test/httpclient_test.py:693: error: Argument 1 to "assertGreaterEqual" of "TestCase" has incompatible type "Optional[float]"; expected "Union[SupportsDunderGE, SupportsDunderLE]"
> + tornado/test/httpclient_test.py:694: error: Argument 1 to "assertLess" of "TestCase" has incompatible type "Optional[float]"; expected "Union[SupportsDunderLT, SupportsDunderGT]"
> 
> psycopg (https://github.com/psycopg/psycopg)
> + tests/test_psycopg_dbapi20.py:43: error: Unused "type: ignore" comment
> + tests/test_psycopg_dbapi20.py:44: error: Unused "type: ignore" comment
> ```

These all look good to me.

- `tornado` source code here: https://github.com/tornadoweb/tornado/blob/4470d6927a598b2d8e7aaf14398ab7ca5b0ed5c6/tornado/test/httpclient_test.py#L693-L694. You can see here that `request_time` can clearly be `None`: https://github.com/tornadoweb/tornado/blob/4470d6927a598b2d8e7aaf14398ab7ca5b0ed5c6/tornado/httpclient.py#L668.
- `psycopg` source code here: https://github.com/psycopg/psycopg/blob/3acc1d2136a7a7b0276c53c23d789cc3a0b8dd6f/tests/test_psycopg_dbapi20.py#L43-L44

**Review by srittau — APPROVED** [review]

**Comment by srittau** [comment]
Thanks!

**Pull request merged**